### PR TITLE
Add Timeout support

### PR DESCRIPTION
**Pull request by Herobone**
I added an short implementation to support timeouts.
All checks passed (running `_example/main.go` and a custom modification of it to support Timed pings)

Added the new `NewTimed` method to not overwrite the existing method

### CODE REVIEW

**Comment by Herobone** [comment]
Added examples and removed Debug

**Comment by raqbit** [comment]
Thank you!

Looks great over all, but I'm wondering if perhaps `context.Context` could be used instead, which would give mc-pinger timeout support for "free" while also adding support for other ways to cancel the ping. 

What do you think?

**Comment by Herobone** [comment]
changed from `net.DialTimeout` to a use of `context.WithTimeout`. I don't know if that's what you wanted but I think it comes close :-)

**Comment by raqbit** [comment]
I was thinking that perhaps a `PingContext` function could accept a context, where the user of the api can pass their own context in which could have a timeout or be tied to some other deadline. This would be in line with how many other apis (Like `Dial` vs `DialContext`) work.

**Comment by Herobone** [comment]
The `DialContext` Method is not officially exposed, but can be called if you use your own `var d Dial`. But I get what you mean.
I will try to implement it later

**Comment by Herobone** [comment]
Added a new Constructor that accepts a Context. Function with just the Timeout still works
Added a new Example for how to use it with Context

File: chat.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"encoding/json"
 )
 
-// Minecraft chat component
+// RegularChatComponent is aMinecraft chat component
```
Comment (raqbit) [inline chat.go]:
```suggestion
// RegularChatComponent is a Minecraft chat component
```

**Review by raqbit — CHANGES_REQUESTED** [review]
One small typo, rest looks great. :+1: 

I do wonder though, with the context version, why there is a need for a separate timeout version if you can simply pass in a context with timeout.

**Comment by Herobone** [comment]
Fixed the typo
I added a separate Timeout because sometimes there is no Context Provided. This saves code when implementing the API

**Review by raqbit — APPROVED** [review]
LGTM :tada:

**Comment by raqbit** [comment]
Thank you for working on this @Herobone :)

**Pull request merged**